submissions/VolodymyrRutskyi/task_js_oop/app.js

```diff
@@ -0,0 +1,101 @@
+const fieldWidth = 510,
+      startX = 202,
+      startY = 405,
+      tileWidth = 100,
+      tileHeight = 83;
+
+class Enemy {
+  constructor(y, speed) {
+    this.x = 0;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+  }
+
+  checkCollisions(){
+    if (
+      player.x < this.x + tileWidth &&
+      player.x + tileWidth > this.x &&
+      player.y < this.y + tileHeight &&
+      tileHeight + player.y > this.y
+    ) {
+      player.x = startX;
+      player.y = startY;
+    }
+  }
+
+  update(dt) {
+    this.x += this.speed * dt;
+    if (this.x > fieldWidth) {
+      this.x = 0;
+    } else {
+      this.checkCollisions();
+    }
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+}
+
+class Player {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+    this.sprite = 'images/char-boy.png';
+  }
+
+  update(dt) {
+    if (this.y < 0) {
+      this.resetPosition();
+    }
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+
+  handleInput(keyPress) {
+    if (keyPress == "left" && this.x > 0) {
+      this.x -= tileWidth;
+    }
+    if (keyPress == "right" && this.x < startY) {
+      this.x += tileWidth;
+    }
+    if (keyPress == "up" && this.y > 0) {
+      this.y -= tileHeight;
+    }
+    if (keyPress == "down" && this.y < startY) {
+      this.y += tileHeight;
+    }
+    if (this.y < 0) {
+      setTimeout(function () {
+        resetPosition();
+      }, 600);
+    }
+  }
+
+  resetPosition() {
+    this.x = startX;
+    this.y = startY;
+  }
+}
+
+const player = new Player(startX, startY),
+      enemyOne = new Enemy (66, 80),
+      enemyTwo = new Enemy (150, 100),
+      enemyTree = new Enemy (236, 120),
```

Comment:
(1) What are these mysterious numbers?
You were supposed to fix code in accordance with the requirements.

* Requirements re **Constants**:
    * [ ] all numbers like block dimensions, initial locations are defined as named constants (e.g. `const STEP = 101;`) as otherwise numbers scattered across code base look cryptic; named constants add semantic meaning and improve readability
    * [ ] every number that has a semantic purpose (like those listed above) should be defined as constants; think of how your code **reads** - the closer to plain English the better
    * [ ] there are core constants and derived constants
      (e.g. derived constant `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
    * [ ] arrays of constants are also constants
      (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
    * [ ] const objects help organizing and structure const data even better
      (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`

(2) DRY
As a peer developer I am tasked to add 10 more enemies.
I want to do this with a minumal effort.